dart/lib/src/protocol/sentry_span.dart

```diff
@@ -41,6 +45,15 @@ class SentrySpan extends ISentrySpan {
     if (_throwable != null) {
       _hub.setSpanContext(_throwable, this, _tracer.name);
     }
+    finishedCallback?.call();
+    await super.finish(status: status);
+  }
+
+  @override
+  void finishAfter(Duration duration, {SpanStatus? status}) {
```

Comment:
ideally this would be replaced by a optional parameter of the startTransaction.

Reply:
If we go this route, we need to provide this param in all `startTransaction` methods throught the framework: `Hub, HubAdapter, ISentrySpan, Sentry`. Think it is less invasive and to do this as method on the span. We could also do casting to `SentryTracer` in the nav observer and only add it there, if we don't want to expose this API.

Reply:
yep, indeed, but that's the idea, so its a feature built-in by the SDK that can be used by custom instrumentation too